### PR TITLE
Implemented toggle of up to two secondary charts

### LINKED ISSUE #187: Add upto 2 secondary charts

### PR DESCRIPTION
**Pull request by djmiley**
Resolves #187

### CODE REVIEW

File: src/assets/js/chart/primaryChart.js [hunk 1]
```diff
@@ -181,7 +181,7 @@
         };
 
         primaryChart.toggleIndicator = function(indicator) {
-            if (indicator.show) {
+            if (indicatorMulti.series().indexOf(indicator.option) === -1) {
                 indicatorMulti.series()
```
Comment (djmiley) [inline src/assets/js/chart/primaryChart.js]: Look at this!

File: src/assets/js/chart/primaryChart.js [hunk 2]
```diff
@@ -44,7 +44,11 @@
...
         var extent = fc.util.extent(visibleData, extentAccessor);
 
-        if (indicators.length) {
+        if (currentIndicators.length) {
+            var indicators = [];
+            for (var i = 0; i < currentIndicators.length; i++) {
+                indicators.push(currentIndicators[i].valueString);
+            }
             var movingAverageShown = (indicators.indexOf('movingAverage') !== -1);
             var bollingerBandsShown = (indicators.indexOf('bollinger') !== -1);
```
Comment (djmiley) [inline src/assets/js/chart/primaryChart.js]: Is there an easier way to test on a shared property of an array of objects?

File: src/assets/js/chart/primaryChart.js [hunk 3]
```diff
@@ -44,7 +44,11 @@
         }
         var extent = fc.util.extent(visibleData, extentAccessor);
 
-        if (indicators.length) {
+        if (currentIndicators.length) {
+            var indicators = [];
```
Comment (jleft) [inline src/assets/js/chart/primaryChart.js]:
```js
 var indicators = currentIndicators.map(function(indicator) { return indicator.valueString; });
```

File: src/assets/js/chart/primaryChart.js (hunk 2 above)
Comment (jleft) [inline src/assets/js/chart/primaryChart.js]: Not sure what you mean?

File: src/assets/js/chart/primaryChart.js (hunk 3 above)
Comment (djmiley) [inline src/assets/js/chart/primaryChart.js]: Placed my line note in the wrong position. Thanks!

File: src/assets/js/chart/primaryChart.js [hunk 4]
```diff
@@ -102,14 +107,12 @@
...
-        var indicatorStrings = [];
-
         function updateMultiSeries() {
-            if (indicatorMulti.series()) {
-                multi.series([gridlines, currentSeries.option, closeLine, indicatorMulti]);
-            } else {
-                multi.series([gridlines, currentSeries.option, closeLine]);
+            multi.series([gridlines, currentSeries.option, closeLine]);
+            if (currentIndicators.length) {
+                for (var i = 0; i < currentIndicators.length; i++) {
```
Comment (jleft) [inline src/assets/js/chart/primaryChart.js]: Could use array [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) again here?

File: src/assets/styles/primary.less [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.primary-row {
+  width: 100%;
```
Comment (jleft) [inline src/assets/styles/primary.less]: Do we need this? Doesn't `.row` have a width of 100%?

File: src/assets/styles/secondary.less [hunk 6]
```diff
@@ -1,3 +1,7 @@
-#secondary-row > .col-chart {
+.secondary-row {
+  width: 100%;
```
Comment (jleft) [inline src/assets/styles/secondary.less]: Do we need this? Doesn't `.row` have a width of 100%?

File: src/assets/js/menu/generator/toggleGroup.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                    value: function(d) { return d.valueString; }
+                });
+        }
+
+        function toggleGenerator(selection) {
+            selection.call(layoutButtons);
+
+            selection.selectAll('.btn')
+                .on('click', function() {
+                    var toggledOption = d3.select(this)
```
Comment (jleft) [inline src/assets/js/menu/generator/toggleGroup.js]: Could we get the `checked` property from the `input` and send that in the event, so we know if the option has been toggled on or off?

**Comment by jleft** [comment]
:+1:

**Pull request merged**